src/getMachines.py

```diff
@@ -0,0 +1,54 @@
+import boto3
+from decEncoder import *
+from boto3.dynamodb.conditions import Key, Attr
+
+"""Scan dynamo table"""
+dynamodb = boto3.resource('dynamodb', region_name = 'us-east-2')
+machineTable = dynamodb.Table('Machines')
+
+"""Lambda handler function for /buildings/{buildingId}/machines API call
+   Returns all the machines in a given building
+   or machines matching query parameters (machineType and/or status)
+   or 'Building or machine not found' error if no machines are returned matching criteria
+"""
+def getMachines(event, context):
+    buildingId = None
+    if event.get('pathParameters'):
+        buildingId = event.get('pathParameters').get('buildingId')
+    if buildingId:
+        machineType = None
+        status = None
+        if event.get('queryStringParameters'):
```

Comment:
Mentioned in our meeting there was a better way of doing this. Sort of like the following:

    filterExpression = Attr('buildingId').eq(buildingId)
    acceptedFilters = ['machineType', 'status']
    if event.get('queryStringParameters'):
        for key in event.get('queryStringParameters'):
            if key in acceptedFilters:
                filterExpression = filterExpression & Attr(key).eq(event.get('queryStringParameters').get(key))
                
Something about that code, you're gonna need to make sure the type of what gets put into the filter expression is correct, otherwise you'll either get nothing or an error. Haven't done any testing about that at this point. Might want to make a function that takes in the key and the value and returns the value cast to the right type.

Also, formatting isn't necessarily the greatest. Feel free to see what you can do to make it nicer.